ArchiSteamFarm/Core/Utilities.cs

```diff
@@ -367,72 +460,83 @@ public static class Utilities {
+				targetBackupDirectory = backupDirectory;
+			}
 
-		// We can now get rid of directories that are empty
-		DeleteEmptyDirectoriesRecursively(targetDirectory);
+			string targetBackupFile = Path.Combine(targetBackupDirectory, fileName);
 
-		if (!Directory.Exists(targetDirectory)) {
-			Directory.CreateDirectory(targetDirectory);
+			File.Move(file, targetBackupFile, true);
 		}
 
-		// Now enumerate over files in the zip archive, skip directory entries that we're not interested in (we can create them ourselves if needed)
-		foreach (ZipArchiveEntry zipFile in zipArchive.Entries.Where(static zipFile => !string.IsNullOrEmpty(zipFile.Name))) {
-			string file = Path.GetFullPath(Path.Combine(targetDirectory, zipFile.FullName));
+		// Finally, we can move the newly extracted files to target directory
+		foreach (string file in Directory.EnumerateFiles(updateDirectory, "*", SearchOption.AllDirectories)) {
+			string fileName = Path.GetFileName(file);
 
-			if (!file.StartsWith(targetDirectory, StringComparison.Ordinal)) {
-				throw new InvalidOperationException(nameof(file));
+			if (string.IsNullOrEmpty(fileName)) {
+				throw new InvalidOperationException(nameof(fileName));
 			}
 
-			if (File.Exists(file)) {
-				// This is possible only with files that we decided to leave in place during our backup function
-				string targetBackupFile = $"{file}.bak";
+			string relativeFilePath = Path.GetRelativePath(updateDirectory, file);
 
-				File.Move(file, targetBackupFile, true);
+			if (string.IsNullOrEmpty(relativeFilePath)) {
+				throw new InvalidOperationException(nameof(relativeFilePath));
 			}
 
-			// Check if this file requires its own folder
-			if (zipFile.Name != zipFile.FullName) {
-				string? directory = Path.GetDirectoryName(file);
+			string? relativeDirectoryName = Path.GetDirectoryName(relativeFilePath);
 
-				if (string.IsNullOrEmpty(directory)) {
-					ASF.ArchiLogger.LogNullError(directory);
+			if (relativeDirectoryName == null) {
+				throw new InvalidOperationException(nameof(relativeDirectoryName));
+			}
 
-					return false;
-				}
+			// We're going to move this file out of the current place, overwriting existing one if needed
+			string targetUpdateDirectory;
 
-				if (!Directory.Exists(directory)) {
-					Directory.CreateDirectory(directory);
-				}
+			if (relativeDirectoryName.Length > 0) {
+				// File inside a subdirectory
+				targetUpdateDirectory = Path.Combine(targetDirectory, relativeDirectoryName);
 
-				// We're not interested in extracting placeholder files (but we still want directories created for them, done above)
-				switch (zipFile.Name) {
-					case ".gitkeep":
-						continue;
-				}
+				Directory.CreateDirectory(targetUpdateDirectory);
+			} else {
+				// File in root directory
+				targetUpdateDirectory = targetDirectory;
 			}
 
-			zipFile.ExtractToFile(file);
+			string targetUpdateFile = Path.Combine(targetUpdateDirectory, fileName);
+
+			File.Move(file, targetUpdateFile, true);
```

Comment:
Duplicates perfectly the code for moving old files to the backup directory, might be worth to make this a separate function